### PR TITLE
set FailureActions for docker, kubelet, kubeproxy on windows

### LINKED ISSUE #3888: Make kubelet & docker service restart automatically when it crashes on Windows
**Is this a request for help?**:
FEATURE REQUEST
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):

---

**What version of acs-engine?**:
v0.21.2
---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes

**What happened**:
I found kubelet on Windows crashes due to OOM, while after that kubelet won't restart automatcially even there is memory later (while on Linux node, it does). Could we configure `kubelet` and `docker` service restart if there is failure like what it does on Linux node. Below are the manual step to restart kubelet:
```
start-service docker
start-service kubeproxy
start-service kubelet
```

**What you expected to happen**:

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:

@PatrickLang @feiskyer

### PR DESCRIPTION
**Pull request by andyzhangx**
**What this PR does / why we need it**:
Original `FailureActions` setting of services `docker`, `kubelet`, `kubeproxy` are not set on Windows, so once service is down, it will never started, user needs to restart VM to make service running again, this PR set `FailureActions` as:
restart after 60000ms (1 minute) for the first, second, and third failures. Reset the counter after 86400 seconds (1 day)

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #3888

**Special notes for your reviewer**:
With this PR, after windows node set up:
```
PS C:\Users\azureuser> get-itemproperty hklm:\system\currentcontrolset\services\kubeproxy

Type                             : 16
Start                            : 2
ErrorControl                     : 1
ImagePath                        : c:\k\nssm.exe
DisplayName                      : Kubeproxy
ObjectName                       : LocalSystem
DelayedAutostart                 : 0
FailureActionsOnNonCrashFailures : 1
DependOnService                  : {Kubelet}
Description                      : Kubeproxy
FailureActions                   : {128, 81, 1, 0...}
PSPath                           : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\system\currentcontrolset\serv
                                   ices\kubeproxy
PSParentPath                     : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\system\currentcontrolset\serv
                                   ices
PSChildName                      : kubeproxy
PSDrive                          : HKLM
PSProvider                       : Microsoft.PowerShell.Core\Registry



PS C:\Users\azureuser> get-itemproperty hklm:\system\currentcontrolset\services\kubelet

Type                             : 16
Start                            : 2
ErrorControl                     : 1
ImagePath                        : c:\k\nssm.exe
DisplayName                      : Kubelet
ObjectName                       : LocalSystem
DelayedAutostart                 : 0
FailureActionsOnNonCrashFailures : 1
Description                      : Kubelet
FailureActions                   : {128, 81, 1, 0...}
PSPath                           : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\system\currentcontrolset\serv
                                   ices\kubelet
PSParentPath                     : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\system\currentcontrolset\serv
                                   ices
PSChildName                      : kubelet
PSDrive                          : HKLM
PSProvider                       : Microsoft.PowerShell.Core\Registry



PS C:\Users\azureuser> get-itemproperty hklm:\system\currentcontrolset\services\docker

Type           : 16
Start          : 2
ErrorControl   : 1
ImagePath      : "C:\Program Files\Docker\dockerd.exe" --run-service
ObjectName     : LocalSystem
FailureActions : {128, 81, 1, 0...}
PSPath         : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\system\currentcontrolset\services\docker
PSParentPath   : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\system\currentcontrolset\services
PSChildName    : docker
PSDrive        : HKLM
PSProvider     : Microsoft.PowerShell.Core\Registry
```

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```
set FailureActions for docker, kubelet, kubeproxy on windows
```

@PatrickLang could you review it first, thanks.
cc @adelina-t @feiskyer

### CODE REVIEW

**Comment by feiskyer** [comment]
> reset the counter after 86400 seconds (1 day)

Could we set this value to something smaller, e.g. 10 min? If kubelet stopped because of something wrong in itself, it may be fixed by restarting kubelet itself (e.g. eviction or garbage collection). So I think it's better to try more times on failure.

By the way, kubelet restart policy is always on Linux.

**Comment by PatrickLang** [comment]
Yes, I agree with @feiskyer . Can you set `reset= 900`  (15 minutes) ?

If it's restarting repeatedly, that means it would:
- 1st restart after 1 minute
- 2nd restart after 1 minute
- 3rd restart after 1 minute
- wait 12 minutes
- restart again ...

If there's a resource leak causing the service to go down - having a 12 minute window would be good so that we can connect, and manually stop the service to prevent restart. Then you can start the service under a debugger instead.

File: parts/k8s/kuberneteswindowssetup.ps1 [hunk 1]
```diff
@@ -674,6 +674,14 @@ Set-Explorer
     New-ItemProperty -Path HKLM:"\\SOFTWARE\\Policies\\Microsoft\\Internet Explorer\\Main" -Name "Start Page" -Type String -Value http://bing.com
 }
 
+function
+Update-ServiceFailureActions()
+{
+    sc.exe failure "kubelet" actions= restart/60000/restart/60000/restart/60000 reset= 86400
+    sc.exe failure "kubeproxy" actions= restart/60000/restart/60000/restart/60000 reset= 86400
+    sc.exe failure "docker" actions= restart/60000/restart/60000/restart/60000 reset= 86400
```
Comment (PatrickLang) [inline parts/k8s/kuberneteswindowssetup.ps1]: Can you update to reset= 900 ?

**Comment by andyzhangx** [comment]
@PatrickLang updated commit, PTAL, thanks.

**Comment by andyzhangx** [comment]
ping @jackfrancis

**Pull request merged**